### DISCUSSION TITLE
Idea: A way to get nearby characters in addLevel()

### DISCUSSION THREAD

**Discussion by sanvishal** [replit/kaboom#267] [Ideas]
I've been using kaboom for a while and I LOVE it. I just have an idea to improve addLevel(). I am not sure if this feature is present and I missed it, but here goes.

```js
const map = addLevel(
  [
    "tttttttttt",
    "t        t",
    "t        t",
    "t        t",
    "t        t",
    "t      $ t",
    "t        t",
    "t $      t",
    "tttttttttt",
  ],
  {
    width: 16,
    height: 16,
    t: ({x,y}) => {
      /*
       * x and y would give me the current character's index in 2D level array
       * using that I would conditionally render sprites
       */
    },
  }
);
```

I could get the nearby characters with the current x and y index to render specific sprites on the current position, this could come in handy for _[auto tiling](https://gamedev.stackexchange.com/questions/46594/elegant-autotiling)_ and procedurally generated levels.

 _lemme know your thoughts on this :)_

**Comment by slmjkdbtl** [comment]
Sorry missed this one. The parameter should already be there!
